lib/sneakers/worker.rb

```diff
@@ -61,11 +61,11 @@ def do_work(delivery_info, metadata, msg, handler)
           end
         rescue Timeout::Error
           res = :timeout
-          worker_error('timeout')
+          worker_error('timeout',msg)
         rescue => ex
           res = :error
           error = ex
-          worker_error('unexpected error', ex)
+          worker_error('unexpected error', msg, ex)
```

Comment:
Sidekiq handles this well for us. When there's an error in a worker, it raises the error including the context of the exception, which has the worker arguments (Sidekiq retires the job then). We use Sentry to capture unhandled exceptions which detects and redacts any sensitive information in the context. This does _not_ add errors or context to the logs, keeping our logs from filling up or leaking sensitive data.

It hasn't mattered as much for us that sneakers doesn't handle this well since we use [`la_gear`](https://github.com/giftcardzen/la_gear) which passes messages straight to Sidekiq from the `Sneakers::Worker#work method` (see [here](https://github.com/giftcardzen/la_gear/blob/develop/lib/la_gear/worker.rb#L68)). However, we have had rare cases that an error was raised in the `Sneakers::Worker` before the message gets into sidekiq, and we are not notified of these errors because sneakers handlers don't have a way to bubble up errors to Sentry. This is is the reason we want to PR sneakers for bubbling that error somehow.